homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -46,66 +60,106 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the alarm platform."""
-    name = config[CONF_NAME]
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    area_id = config[CONF_AREA_ID]
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Import Yale configuration from YAML."""
+    LOGGER.warning(
+        "Loading Yale Alarm via platform setup is depreciated; Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
 
-    try:
-        client = YaleSmartAlarmClient(username, password, area_id)
-    except AuthenticationError:
-        _LOGGER.error("Authentication failed. Check credentials")
-        return
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the alarm entry."""
 
-    add_entities([YaleAlarmDevice(name, client)], True)
+    async_add_entities(
+        [YaleAlarmDevice(coordinator=hass.data[DOMAIN][entry.entry_id]["coordinator"])]
+    )
 
 
-class YaleAlarmDevice(AlarmControlPanelEntity):
+class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
-    def __init__(self, name, client):
-        """Initialize the Yale Alarm Device."""
-        self._name = name
-        self._client = client
-        self._state = None
+    _state_map = {
+        YALE_STATE_DISARM: STATE_ALARM_DISARMED,
+        YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
+        YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
+    }
 
-        self._state_map = {
-            YALE_STATE_DISARM: STATE_ALARM_DISARMED,
-            YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
-            YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
-        }
+    _state = STATE_UNAVAILABLE
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self.coordinator.entry.data[CONF_NAME]
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return str(self.coordinator.entry.entry_id)  # type: ignore[attr-defined]
```

Comment:
We should type the coordinator class attribute so we don't need to ignore typing.

Reply:
On this one too I am not sure what you mean. Can you provide an example?

Reply:
Ok, got this one sorted too

Reply:
Nope, no idea so need help on this one. Thanks

Reply:
Think I got it sorted but have a look